tokio/src/runtime/scheduler/multi_thread/worker.rs

```diff
@@ -1090,17 +1082,15 @@ impl Handle {
             true
         } else {
             // Push to the LIFO slot
-            let prev = core.lifo_slot.take();
-            let ret = prev.is_some();
+            let prev = core.run_queue.push_lifo(task);
 
             if let Some(prev) = prev {
                 core.run_queue
                     .push_back_or_overflow(prev, self, &mut core.stats);
+                true
+            } else {
+                false
             }
-
-            core.lifo_slot = Some(task);
-
-            ret
         };
```

Comment:
I took a longer time than the old code to understand the new one. Is `match core.run_queue.push_lifo(task)` more readable?